src/test/java/org/terasology/genome/GenomeNewTest.java

```diff
@@ -0,0 +1,150 @@
+import org.terasology.world.WorldProvider;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * A test to check working of multiple traits with different breeding algorithms. Addition of multiple breeding
+ * algorithms was based on the structure of these tests.
+ */
+class GenomeNewTest {
+    @In
+    WorldProvider worldProvider;
+
+    @In
+    GenomeRegistry genomeRegistry;
+
+    GeneMutator geneMutator;
+    BreedingAlgorithm defaultBreedingAlgorithm;
+    float mutationChance;
+
+    @BeforeEach
+    public void setUp() {
+        //A=5, F=10, K=15
+        geneMutator = new VocabularyGeneMutator("ABCDEFGHIJK");
+        mutationChance = 0.05f;
+        defaultBreedingAlgorithm = new MonoploidBreedingAlgorithm(0, mutationChance, geneMutator);
+    }
+
+    @Test
+    void testContinuousTrait() {
+        GeneIndexGenomeMap genomeMap = new GeneIndexGenomeMap();
+        int geneIndices[] = new int[]{0};
+        genomeMap.addProperty("filling", geneIndices, Integer.class, defaultBreedingAlgorithm,
+                new Function<String, Integer>() {
+                    @Nullable
+                    @Override
+                    public Integer apply(@Nullable String input) {
+                        return (input.charAt(0) - 'A' + 5);
+                    }
+                });
+
+        GenomeDefinition genomeDefinition = new GenomeDefinition(defaultBreedingAlgorithm, genomeMap);
+        GeneIndexGenomeMap genomeMap1 = (GeneIndexGenomeMap) genomeDefinition.getGenomeMap();
+        assertEquals(genomeMap1.propertyDefinitionMap.toString(), genomeMap.propertyDefinitionMap.toString());
+        System.out.println(genomeMap1.getProperty("filling", "Aa", Integer.class));
+    }
+
+    @Test
+    void testDiscreteTrait() {
+        GeneIndexGenomeMap genomeMap = new GeneIndexGenomeMap();
+        int geneIndices[] = new int[]{0, 1};
+        genomeMap.addProperty("height", geneIndices, Integer.class, defaultBreedingAlgorithm,
+                new Function<String, Integer>() {
+                    @Nullable
+                    @Override
+                    public Integer apply(@Nullable String input) {
+                        return (Character.isUpperCase(input.charAt(0)) ? 1 : 0);
+                    }
+                });
+
+        GenomeDefinition genomeDefinition = new GenomeDefinition(defaultBreedingAlgorithm, genomeMap);
+        GeneIndexGenomeMap genomeMap1 = (GeneIndexGenomeMap) genomeDefinition.getGenomeMap();
+        assertEquals(genomeMap1.propertyDefinitionMap.toString(), genomeMap.propertyDefinitionMap.toString());
+        assertEquals(genomeMap1.getProperty("height", "TT", Integer.class), 1);
+        assertEquals(genomeMap1.getProperty("height", "tt", Integer.class), 0);
+        System.out.println(genomeMap1.getProperty("height", "Tt", Integer.class));
```

Comment:
Is there a specific reason for these outputs to console? If these are for confirming something, they can be converted to another assert statement?

Reply:
Fixed